dev/devicelab/lib/framework/running_processes.dart

```diff
@@ -0,0 +1,228 @@
+
+Future<bool> killProcess(int pid, {ProcessManager processManager}) async {
+  assert(pid != null, 'Must specify a pid to kill');
+  processManager ??= const LocalProcessManager();
+  ProcessResult result;
+  if (Platform.isWindows) {
+    result = await processManager.run(<String>[
+      'taskkill',
+      '/pid', pid.toString(),
+      '/f',
+    ]);
+  } else {
+    result = await processManager.run(<String>[
+      'kill',
+      '-9',
+      pid.toString(),
+    ]);
+  }
+  return result.exitCode == 0;
+}
+
+Stream<RunningProcessInfo> getRunningProcesses({
+  String processName,
+  ProcessManager processManager,
+}) {
+  processManager ??= const LocalProcessManager();
+  if (Platform.isWindows) {
+    return windowsRunningProcesses(processName, processManager);
+  }
+  return posixRunningProcesses(processName, processManager);
+}
+
+@visibleForTesting
+Stream<RunningProcessInfo> windowsRunningProcesses(
+    String processName, ProcessManager processManager) async* {
+  final String script = processName != null
+      ? '''"Get-CimInstance Win32_Process -Filter \\"name='$processName'\\" | Select-Object ProcessId,CreationDate,CommandLine | Format-Table -AutoSize | Out-String -Width 4096"'''
+      : '"Get-CimInstance Win32_Process | Select-Object ProcessId,CreationDate,CommandLine | Format-Table -AutoSize | Out-String -Width 4096"';
+  final ProcessResult result = await processManager.run(<String>[
+    'powershell',
+    '-c',
+    script,
+  ]);
+  if (result.exitCode != 0) {
+    print('Could not list processes!');
+    print(result.stderr);
+    print(result.stdout);
+    exit(result.exitCode);
+  }
+  for (RunningProcessInfo info in processPowershellOutput(result.stdout)) {
+    yield info;
+  }
+}
+
+@visibleForTesting
+Iterable<RunningProcessInfo> processPowershellOutput(String output) sync* {
+  // e.g.:
+  //
+  // ProcessId CreationDate          CommandLine
+  // --------- ------------          -----------
+  //      2904 3/11/2019 11:01:54 AM "C:\Program Files\Android\Android Studio\jre\bin\java.exe" -Xmx1536M -Dfile.encoding=windows-1252 -Duser.country=US -Duser.language=en -Duser.variant -cp C:\Users\win1\.gradle\wrapper\dists\gradle-4.10.2-all\9fahxiiecdb76a5g3aw9oi8rv\gradle-4.10.2\lib\gradle-launcher-4.10.2.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 4.10.2
+  if (output == null) {
+    return;
+  }
+
+  const int processIdHeaderSize = 'ProcessId'.length;
+  const int creationDateHeaderStart = processIdHeaderSize + 1;
+  int creationDateHeaderEnd;
+  int commandLineHeaderStart;
+  bool inTableBody = false;
+  for (String line in output.split('\n')) {
+    if (line.startsWith('ProcessId')) {
+      commandLineHeaderStart = line.indexOf('CommandLine');
+      creationDateHeaderEnd = commandLineHeaderStart - 1;
+    }
+    if (line.startsWith('--------- ------------')) {
+      inTableBody = true;
+      continue;
+    }
+    if (!inTableBody || line.isEmpty) {
+      continue;
+    }
+
+    DateTime getDateTime() {
```

Comment:
I don't see the utility of the local function. If you're worried about name collisions, you can use braces to create a block:

```
final a = '';
{
  final a = '';
}
```

Reply:
I had it this way when I was parsing lines in a loop differently.  No need for it now.